Demo/src/main/java/com/paypal/android/api/model/PaymentSource.kt

```diff
@@ -0,0 +1,39 @@
+package com.paypal.android.api.model
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+import org.json.JSONObject
+
+@Parcelize
+data class PaymentSource(
+    val paypal: PayPalDetails? = null
+) : Parcelable {
+    companion object {
+
+        internal fun fromJson(jsonObject: JSONObject): PaymentSource? {
```

Comment:
I'm still not a huge fan of `fromJson()` methods because the method itself doesn't say much about where the JSON came from. Can we move this parsing logic somewhere external to this file? It would be nice if this file contained pure data objects without any `companion` logic.